modules/subdyn/src/FEM.f90

```diff
@@ -1157,6 +1165,159 @@ SUBROUTINE ElemK_Cable(A, L, E, T0, DirCos, K)
    K = MATMUL( MATMUL(DC, K), TRANSPOSE(DC) ) ! TODO: change me if DirCos convention is  transposed
 END SUBROUTINE ElemK_Cable
 !------------------------------------------------------------------------------------------------------
+!> Element stiffness matrix for spring
+!! The spring element can include diagnal and cross-coupling positions. 
+!! Assuming that the stiffness is symmetric (21 stiffness coefficients). The stiffness matrix could also be non-symmetric, if desired.  
+SUBROUTINE ElemK_Spring(k11, k12, k13, k14, k15, k16, k22, k23, k24, k25, k26, k33, k34, k35, k36, k44, k45, k46, k55, k56, k66, DirCos, K)
+   REAL(ReKi), INTENT( IN) :: k11, k12, k13, k14, k15, k16, k22, k23, k24, k25, k26, k33, k34, k35, k36, k44, k45, k46, k55, k56, k66
+   REAL(FEKi), INTENT( IN) :: DirCos(3,3) !< From element to global: xg = DC.xe,  Kg = DC.Ke.DC^t
+   REAL(FEKi), INTENT(OUT) :: K(12, 12) 
+   ! Local variables
+   REAL(FEKi)                            :: DC(12, 12)
+   
+   K(1:12,1:12) = 0.0_FEKi
```

Comment:
This might be a silly suggestion, but could the indexing be simplified by building the 6x6 K matrix, then applying the element to global rotation DC (6x6) to it, and finally copying the resulting 6x6 into kg (12x12) four times? This method would be equivalent.

Reply:
Yes, you could rotate the 6x6 K element matrix and then copy the resulting matrix four times.